File: 242-valid-anagram/242-valid-anagram.java
class Solution {
    String sort(String s){
        char c[]=s.toCharArray();
        Arrays.sort(c);
        s=String.valueOf(c);
        return s;
    }
    public boolean isAnagram(String s, String t) {
        if(s.length()!=t.length())
            return false;
        s=sort(s);
        t=sort(t);
        return s.equals(t);
    }
}

File: 242-valid-anagram/242-valid-anagram.cpp
class Solution {
public:
    unordered_map<int,int>m;
    bool isAnagram(string s, string t) {
        if(s.size()!=t.size())
            return false;
        for(char c:s){
            m[c]++;
        }
        for(char c:t){
            m[c]--;
            if(m[c]==0)
                m.erase(c);
        }
        return m.size()==0;
    }
};